src/main/java/application/CaseController.java

```diff
@@ -29,4 +31,14 @@ public CaseFilterResponseBean filterCasesHQ(@RequestBody String body) throws Exc
         String caseResponse = CaseAPIs.filterCases(filterRequest);
         return new CaseFilterResponseBean(caseResponse);
     }
+
+    @RequestMapping("/sync_db")
+    public SyncDbResponseBean syncUserDb(@RequestBody String body) throws Exception {
+        ObjectMapper mapper = new ObjectMapper();
+        SyncDbRequestBean syncRequest = mapper.readValue(body, SyncDbRequestBean.class);
```

Comment:
You shouldn't have to do this yourself, Spring can do it for you if you've registered the [MappingJacksonHttpMessageConverter](http://docs.spring.io/spring/docs/3.0.x/javadoc-api/org/springframework/http/converter/json/MappingJacksonHttpMessageConverter.html) which I think you have via the [@EnableWebMvc](http://docs.spring.io/spring/docs/current/spring-framework-reference/html/mvc.html#mvc-config-enable) annotation:

```java
@RequestMapping("/sync_db")
public SyncDbResponseBean syncUserDb(@RequestBody SyncDbRequestBean syncRequest) throws Exception {
    syncRequest.setRestoreService(restoreService);
    .....
}
```